db/expire.go

```diff
@@ -209,13 +206,19 @@ func doExpire(txn *Transaction, mkey, id []byte) error {
 	obj, err := getObject(txn, mkey)
 
 	// Check for dirty data due to copying or flushdb/flushall
-	if err == ErrKeyNotFound || !bytes.Equal(obj.ID, id) {
+	if err == ErrKeyNotFound {
 		return nil
 	}
-
 	if err != nil {
 		return err
 	}
+	idLen := len(id)
+	if idLen > len(obj.ID) {
+		id = id[:idLen-1]
```

Comment:
Maybe use `len(obj.ID)` as `idLen` is better, we don't need to struggle why the length should be `len-1`

```go
idLen = len(obj.ID)
if len(id) > idLen {
    id = id[:idLen]
}
```

Reply:
It is much more clearer. Thanks.